pkg/dockerregistry/server/pullthroughblobstore.go

```diff
@@ -90,7 +90,7 @@ func (r *pullthroughBlobStore) proxyStat(ctx context.Context, retriever importer
 		return distribution.Descriptor{}, err
 	}
 	pullthroughBlobStore := repo.Blobs(ctx)
-	desc, err := pullthroughBlobStore.Stat(r.repo.ctx, dgst)
+	desc, err := pullthroughBlobStore.Stat(ctx, dgst)
```

Comment:
Looking at the [`Stat()` call](https://github.com/openshift/origin/blob/fcbe87a3d4c4088adc729c2db9ca66964b859ddd/pkg/dockerregistry/server/pullthroughblobstore.go#L31), the only statement that could return `unauthorized` is [`getImageStream()`](https://github.com/openshift/origin/blob/fcbe87a3d4c4088adc729c2db9ca66964b859ddd/pkg/dockerregistry/server/pullthroughblobstore.go#L47) which doesn't use context at all. So I don't think this will help. Sorry for the initial red herring.

Nevertheless, I think that the use of context should be consistent (use `ctx` where possible). Using `r.ctx` in its place should be documented.

Why registry is unauthorized to get the image stream is beyond my understanding.